### PR TITLE
query_string_query should take term length into consideration when fuzziness is auto

### LINKED ISSUE #15972: fuzziness=auto doesn't work in query string query
https://gist.github.com/masaruh/55deca377b6ade5847de
(on 1.x, it's always `~2`)

Probably because it always uses fixed value https://github.com/elastic/elasticsearch/blob/v2.1.1/core/src/main/java/org/apache/lucene/queryparser/classic/MapperQueryParser.java#L141

Probably it should be something like?
```
        if (fuzzySlop.image.length() == 1) {
            return getFuzzyQuery(qfield, termImage, settings.getFuzziness().asDistance(termImage));
        }
```

### PR DESCRIPTION
**Pull request by masaruh**
Fixes #15972

### CODE REVIEW

File: core/src/main/java/org/apache/lucene/queryparser/classic/MapperQueryParser.java [hunk 1]
```diff
@@ -114,7 +114,7 @@ public void reset(QueryParserSettings settings) {
     @Override
     Query handleBareFuzzy(String qfield, Token fuzzySlop, String termImage) throws ParseException {
         if (fuzzySlop.image.length() == 1) {
-            return getFuzzyQuery(qfield, termImage, Float.toString(fuzzyMinSim));
```
Comment (jpountz) [inline core/src/main/java/org/apache/lucene/queryparser/classic/MapperQueryParser.java]: let's remove fuzzyMinSim entirely from this class, it looks unused

**Comment by jpountz** [comment]
Looks good, I left a suggestion for a cleanup.

File: core/src/main/java/org/apache/lucene/queryparser/classic/MapperQueryParser.java (hunk 1 above)
Comment (masaruh) [inline core/src/main/java/org/apache/lucene/queryparser/classic/MapperQueryParser.java]: Right. It's not used. But it's super class's field. I'm going to remove a line setting the value.

**Comment by jpountz** [comment]
LGTM

**Comment by masaruh** [comment]
Thanks @jpountz!

**Pull request merged**